lib/GHCup/Types.hs

```diff
@@ -379,6 +379,7 @@ instance Pretty TarDir where
 -- | Where to fetch GHCupDownloads from.
 data URLSource = GHCupURL
                | StackSetupURL
+               | ChannelAlias  ChannelAlias
```

Comment:
We don't want to touch `URLSource`. It's there for backwards comp and already supports `NewURLSource` through the `SimpleList` constructor.

Reply:
Ok, fixed

Reply:
hmm, the `ghcup -s vanilla install ghc 9.8.3` no longer works without `URLSource`. I should probably convert `optUrlSource   :: Maybe URLSource` to `optUrlSource   :: Maybe NewURLSource`

Reply:
> I should probably convert optUrlSource   :: Maybe URLSource to optUrlSource   :: Maybe NewURLSource

That sounds like a breaking change, no?

Did you check `parseUrlSource` in `GHCup.Utils.Parsers`?

Reply:
I did this in backward compatible way, by modifying `parseUrlSource` too. It now handles both `URLSource` and `NewURLSource`

`cabal run exe:ghcup -- -s "{\"OwnSource\": "\"https://raw.githubusercontent.com/haskell/ghcup-metadata/master/ghcup-prereleases-0.0.8.yaml\""}" list`

`cabal run exe:ghcup -- -s vanilla install ghc 8.10.2`